buildlib/io_demo/iodemo_analyzer.py

```diff
@@ -0,0 +1,248 @@
+import subprocess
+import os
+import argparse
+import re
+import datetime,time
+import traceback,sys
+
+allow_error_list = [
+    'Connection reset by remote peer',
+    'UCX-connection.*detected error:',
+    'ERROR Remote QP on mlx',
+    'UCX  ERROR RC QP',
+    'ERROR IB Async event on',
+    'setting error flag on connection',
+    'Operation rejected by remote peer',
+    'got error event RDMA_CM_EVENT_ADDR_ERROR',
+    'rdma_accept',
+    'UCX  ERROR Remote access on',
+    'UCX  ERROR Transport retry count exceeded on',
+    'UCX  WARN  failed to disconnect CM lane',
+    'ucp_ep_create\(\) failed: Input/output error',
+    'terminate connection.*due to Input/output error',
+    'UCX  ERROR Local QP operation on',
+    'conn_id send request.*failed: Input/output error',
+    'deleting connection with status Input/output error',
+    'UCX  WARN  failed to disconnect CM lane .* Operation rejected by remote peer',
+    'ucp_ep_query\(\) failed: Endpoint timeout',
+    'UCX  ERROR rdma_reject.*failed with error: Invalid argument',
+    'UCX  ERROR rdma_init_qp_attr.*failed: Invalid argument',
+    'UCX  ERROR rdma_establish on ep.*failed: Invalid argument',
+    'UCX  ERROR .*client.*failed to process a connect response'
+    ]
+
+
+re_allow_list = re.compile("|".join(allow_error_list), re.I)
+re_timestamp = re.compile(r"\[(\d+\.\d+)\].*")
+re_traffic = re.compile(r"\[(\d+\.\d+)\].*read (\d+.\d+).*min:(\d+).*write (\d+.\d+).*min:(\d+).*")
+re_traffic_read = re.compile(r"\[(\d+\.\d+)\].*read:(\d+) ops.*")
+re_traffic_write = re.compile(r"\[(\d+\.\d+)\].*write:(\d+) ops.*")
+re_error = re.compile(r".*(error|assert|backtrace|segmentation).*", re.I)
+re_warning = re.compile(r".*warn.*", re.I)
+
+
+def in_allow_list(line, is_allow_list):
+    is_allow_error = False
+    if is_allow_list:
+        s = re_allow_list.search(line)
+        if s:
+            is_allow_error = True
+    return is_allow_error
+
+
+def process_seek(seek_file):
+    data = {}
+    if seek_file and os.path.exists(seek_file):
+        with open(seek_file) as f:
+            for line in f.readlines():
+                if line:
+                    # [log name] \t [position] \t [previous  timestamp]
+                    d = line.split('\t')
+                    ts = datetime.datetime.fromtimestamp(float(d[2]))
+                    rx_ts = float(d[3])
+                    tx_ts = float(d[4])
+                    data[d[0]] = {
+                        'pos': int(d[1]),
+                        'timestamp': ts,
+                        'timestamp_rx': rx_ts,
+                        'timestamp_tx': tx_ts,
+                        }
+        # Burn After Reading
+        open(seek_file, 'w').close()
+    return data
+
+def read_file(filename):
+    data = ""
+    with open(filename) as f:
+        data = f.readlines()
+    return data
```

Comment:
can we reuse this function in process_seek?

Reply:
it is unused function - removed